nano/node/blockprocessor.cpp

```diff
@@ -42,10 +42,20 @@ void nano::block_processor::flush ()
 	}
 }
 
-bool nano::block_processor::full ()
+size_t nano::block_processor::size ()
 {
 	std::unique_lock<std::mutex> lock (mutex);
-	return (blocks.size () + state_blocks.size ()) > node.flags.block_processor_full_size;
+	return (blocks.size () + state_blocks.size () + forced.size ());
+}
+
+bool nano::block_processor::full ()
+{
+	return size () > node.flags.block_processor_full_size;
+}
+
+bool nano::block_processor::half_full ()
```

Comment:
I think call sites would be easier to read if this was called `above_half_full`, I had to check the relational operator a couple of times on my first read-through.